### ISSUE TITLE
False positive for useless-suppression with cyclic-import

### ISSUE THREAD

**Issue by efroemling** [PyCQA/pylint#3064]
I have a number of import cycles in my codebase, though I try to be safe about them by keeping them in functions and not at the module level.  In this case I suppress the cyclic-import warning within those individual functions.  I just enabled useless-suppression detection, however, and found that it is reporting (seemingly all?) of these cyclic-import suppressions as unnecessary, though when I remove them I once again get cyclic-import warnings.

### Steps to reproduce
1. create a.py with the following code:
```
"""Test Module A."""
import b

print(b)
```

2. create b.py with the following code:
```
"""Test Module B."""

def bfunc():
    """Create a 'safe-ish' import cycle by doing it from a function."""
    import a
    print(a)
```
3. pylint --enable=cyclic-import a.py b.py -> should generate a warning
4. add  # pylint: disable=cyclic-import under bfunc()
5. pylint --enable=cyclic-import a.py b.py -> no warning now
6. pylint --enable=cyclic-import,useless-suppression a.py b.py -> incorrectly states that the line we added in step 4 is unnecessary

### Current behavior
I'm finding all of my cyclic-import suppressions are generating these warnings.

### Expected behavior
Should useless-suppression detection simply ignore cyclic-import suppressions?  Even if it was behaving as expected in this test case, would it still trigger incorrectly if only a.py or b.py were tested by itself?  (pylint only generates the cycle error when both a.py and b.py are tested together)

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.4 (default, Jul  9 2019, 18:13:23) 
[Clang 10.0.1 (clang-1001.0.46.4)]

**Comment by PCManticore** [comment]
Thanks for creating the issue. Most likely `useless-supression` should be ignored for `cyclic-import`, especially since the latter is done after a full program analysis, while the former might not have that context just yet.
